### PR TITLE
Add Inactive Jfrog platform handling (free-tier)

### PR DESCRIPTION
**Pull request by talarian1**
- [ ] All [tests](https://github.com/jfrog/jfrog-idea-plugin/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
-----
Detect and suggest a solution for an inactive JFrog platform (free-tier).

Screenshot 2023-02-06 at 11 25 03

Fixes https://github.com/jfrog/jfrog-idea-plugin/issues/228
Depends on https://github.com/jfrog/xray-client-java/pull/42, https://github.com/jfrog/build-info/pull/698

### CODE REVIEW

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 1]
```diff
@@ -84,13 +93,36 @@ public void startScan() {
...
+        } catch (JFrogInactiveEnvironmentException e) {
+            handleJfrogInactiveEnvironment(e.getRedirectUrl());
         } catch (IOException | RuntimeException | InterruptedException e) {
             logError(Logger.getInstance(), "", e, true);
         } finally {
             executor.shutdown();
         }
     }
 
+    private void handleJfrogInactiveEnvironment(String redirectUrl) {
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:105]: Let's add a JavaDoc above this method explaining what it does

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 2]
```diff
@@ -84,13 +93,36 @@ public void startScan() {
...
+                        })
+                .setCloseButtonEnabled(true)
+                .setHideOnAction(true)
+                .setHideOnClickOutside(true)
+                .setHideOnLinkClick(true)
+                .setHideOnKeyOutside(true)
+                .setDialogMode(true)
+                .createBalloon();
+        balloon.show(RelativePoint.getNorthWestOf(statusBar.getComponent()), Balloon.Position.atRight);
+
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:123]: Redundant new line
```suggestion
```

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java [hunk 3]
```diff
@@ -84,13 +93,36 @@ public void startScan() {
                     logError(Logger.getInstance(), "", e, true);
                 }
             }
+        } catch (JFrogInactiveEnvironmentException e) {
+            handleJfrogInactiveEnvironment(e.getRedirectUrl());
```
Comment (yahavi) [inline src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java:97]: Shouldn't we do this also in the test connection button?
Same question for CI mode.

**Review by yahavi — CHANGES_REQUESTED** [review]

File: src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java (hunk 3 above)
Comment (talarian1) [reply src/main/java/com/jfrog/ide/idea/scan/ScanManagersFactory.java]: Added this to the test connection:
Screenshot 2023-02-19 at 18 15 56
Regarding the CI view, I don't think it is necessary - an Error will be displayed in the logs.

**Pull request merged**